ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala

```diff
@@ -19,9 +54,26 @@ object Logging {
       .getOrElse(throw new java.lang.IllegalArgumentException(s"Unknown logging level $level"))
   }
 
+  @SuppressWarnings(Array("org.wartremover.warts.All"))
+  private implicit val scoptLogEncoder: scopt.Read[LogEncoder] =
+    scopt.Read.reads { encoder =>
+      Either
+        .cond(
+          KnownLogEncoders.contains(encoder.toLowerCase()),
+          encoder.toLowerCase match {
+            case "plain" => LogEncoder.Plain
+            case "json" => LogEncoder.Json
+          },
+          s"Unrecognized logging encoder $encoder",
+        )
+        .getOrElse(
```

Comment:
if you drop the message in the `Left` you might as well use an `Option above or just throw directly in the pattern match.